types/ember__application/index.d.ts

```diff
@@ -119,12 +120,12 @@ export default class Application extends Engine {
  * objects is the responsibility of an "owner", which handled its
  * instantiation and manages its lifetime.
  */
-export function getOwner(object: unknown): unknown;
+export function getOwner(object: unknown): Owner;
```

Comment:
This one is actually kind of annoying. It is actually `Owner | undefined`, because if you ask for the owner of something which hasn’t had `setOwner` called on it, the answer is "nah". @dfreeman and I were talking about this recently: we *could* provide an overload that does the right thing, roughly like this:

```ts
type KnownFrameworkTypes = GlimmerComponent | EmberObject;
export function getOwner(object: KnownFrameworkTypes): Owner;
export function getOwner(object: unknown): Owner | undefined;
```

Then if someone does `getOwner(this)` in anything descended from `EmberObject` *or* `GlimmerComponent`, it will correctly always return `Owner`, and otherwise will return `Owner | undefined` for types Ember itself doesn't explicitly know about. @dfreeman were there any other types that should be in that union if we do this?